PLANNER-2265 Add constraints-drl property (#1065)
kiegroup/optaplanner: optaplanner-docs/src/main/asciidoc/Integration/Integration-chapter.adoc

A classpath resource to read the solver configuration XML.
Defaults to `solverConfig.xml`.
If this property isn't specified, that file is optional.

> optaplanner.constraints-drl::

Reviewer: This is documented in the Spring boot section. We need it copy paste into the Quarkus section too.

Reviewer: Order wise, I suggest we follow the order of SolverConfig.java
So this property should come after optaplanner.solver.move-thread-count and before optaplanner.solver.termination.spent-limit

Author: That's an option of course; I tried to follow the order of fields in the autoconfiguration.

Author: Decision: respect the order of field in the autoconfiguration.

Author: ad the Quarkus section: I was wandering: how could I have missed the quarkus section of our documentation? The answer is quite prosaic: the section is empty so far.
I assigned the https://issues.redhat.com/browse/PLANNER-2284 I filed to myself; not in scope of this PR, though.

> A classpath resource to read the solver score DRL.

Reviewer: Very soft suggestion: I can't explain why, but "solver score DRL" doesn't sound right to me, maybe because it implies there is a difference between "score DRL" and "solver score DRL"? (which there is not). 3 lines, above, "Solver configuration XML" is right. 
```suggestion
A classpath resource to read the score DRL.
```

Defaults to `constraints.drl`.

> Do not define this property when a `ConstraintProvider`, `EasyScoreCalculator` or `IncrementalScoreCalculator` class exists.

Reviewer: Great work. Mind if I add this issue on your backlog (not for this PR)? https://issues.redhat.com/browse/PLANNER-2288

Author: No problem, in fact I have already created one (PLANNER-2284) - see one of my earlier responses on this PR.

> This property must NOT be defined in case a `ConstraintProvider` implementation or other score calculation type exists.

Reviewer: style: "must not be defined" is too passive tense (= too verbose)

```suggestion
Do note define this property when a `ConstraintProvider` implementation or other score calculation type exists.
```

Reviewer: Update: more specific is better:
```suggestion
Do note define this property when a `ConstraintProvider`, `EasyScoreCalculator` or `IncrementalScoreCalculator` class exists.
```
